source/iface/properties.rst

```diff
@@ -2,20 +2,191 @@
   Copyright 2020 The Khronos Group Inc.
   SPDX-License-Identifier: CC-BY-4.0
 
-.. _property_list:
+.. _properties:
 
+**********
+Properties
+**********
+
+Each of the following SYCL runtime classes: ``accessor``, ``buffer``,
+``host_accessor``, ``host_sampled_image_accessor``,
+``host_unsampled_image_accessor``, ``context``,
+``local_accessor``, ``queue``, ``sampled_image``,
+``sampled_image_accessor``, ``stream``, ``unsampled_image``,
+``unsampled_image_accessor`` and ``usm_allocator``
+provide an optional parameter in each of their constructors
+to provide a ``property_list`` which contains zero or more properties.
+Each of those properties augments the semantics of the class with
+a particular feature. Each of those classes must also provide
+``has_property`` and ``get_property`` member functions
+for querying for a particular property.
+
+The example below illustrates how using properties does not affect the
+type of the object, thus, does not prevent the usage of SYCL objects
+in containers.
+
+.. _ buffer-properties-example:
+
+``buffer-properties-example``
+=============================
+
+::
+
+  {
+    context myContext;
+
+    std::vector<buffer<int, 1>> bufferList {
+      buffer<int, 1> { ptr, rng },
+      buffer<int, 1> { ptr, rng, property::use_host_ptr {} },
+      buffer<int, 1> { ptr, rng, property::context_bound { myContext } }
+    };
+
+    for (auto& buf : bufferList) {
+      if (buf.has_property<property::context_bound>()) {
+        auto prop = buf.get_property<property::context_bound>();
+        assert(myContext == prop.get_context());
+      }
+    }
+  }
+
+.. seealso:: |SYCL_BUFF_PROP|
+
+Each property is represented by a unique class and an instance of a property
+is an instance of that type. Some properties can be default constructed while
+others will require an argument on construction. A property may be applicable
+to more than one class, however some properties may not be compatible
+with each other.
+
+.. _property_list`:
+
+``sycl::property_list``
+=======================
+
+Each of the runtime classes mentioned above must provide a common interface of
+member functions in order to fulfill the property interface requirements.
+
+A synopsis of the common properties interface, the SYCL ``property_list``
+class and the SYCL property classes is provided below.
+
+::
+
+  namespace sycl {
```

Comment:
Some code-aligning problems also (in class T especially).
Also API with templates, I think, is better to divide into two lines.